### ISSUE TITLE
drag-and-dropping a file will overwrite without asking to confirm

### ISSUE THREAD

**Issue by Pomax** [mozilla/thimble.mozilla.org#1887]
if I drag-and-drop and index.html from my desktop into the thimble file browser, it immediately overwrites whatever index.html there is already, without asking me whether that is what I meant to do, and without the ability to undo that drop action.

**Comment by Shank09** [comment]
This issue seems interesting, I'd like to work on it 👍

**Comment by humphd** [comment]
Our DnD code lives here: https://github.com/mozilla/brackets/blob/master/src/utils/DragAndDrop.js

**Comment by humphd** [comment]
NOTE: I'm significantly refactoring the dnd code in https://github.com/mozilla/brackets/pull/659.  If you want to add the "don't overwrite existing files" logic to what I'm doing, I can walk you through it.  It wouldn't be a lot of work.  Let me know.

**Comment by humphd** [comment]
@Shank09 are you actually doing this?  If not, @simon66 says he'd like to do it.  Let me know ASAP

**Comment by Shank09** [comment]
@humphd Sure let him take it. I have another bug for Release 0.4, which im currently working on

**Comment by simon66** [comment]
Thanks for the quick reply @Shank09. I'll start working on this asap :+1:

**Comment by simon66** [comment]
Just to recap,
Should a "warning" popup only when a file is being overridden? Or should it be shown every time something is `dragged & dropped`?

Here's the message we can show the user:

"Are you sure you want to import {0} into your project?
This process is irreversible. If unsure, hit Cancel

<Cancel> <Ok>"

Where {0} is the file/zip name.

**Comment by humphd** [comment]
I think we should do something like:

> "A file named {0} already exists in this location. Do you want to use the imported file or keep the existing?"
> `<Use Imported> <Keep Existing> <Cancel>`

**Comment by simon66** [comment]
Perfect! I like the idea.
The current message works fine for single file imports but when an user imports a .zip file, there might be multiple files that could be overridden. So I had this in mind.

If only a single file is being overridden, I'll use this message:

> "A file named {0} already exists in this location. Do you want to use the imported file or keep the existing?"
> <Use Imported> <Keep Existing> <Cancel>
Where {0} is the file name.

And for multiple overrides, I'll use this message:

> "Certain files within {0} already exists in this location. Do you want to use the imported files or keep the existing?"
> <Use Imported> <Keep Existing> <Cancel>
Where {0} is the archive name.

**Comment by humphd** [comment]
I don't think you can separate it out like this, since you'll get `EEXISTS` errors as you try to write the files one-by-one.  Just deal with it at the individual file level, and people can choose.  I might want some, but not all of the files in a zip to overwrite.  This gives the user a choice.

**Comment by humphd** [comment]
I've landed #659 so, feel free to rebase and finish this.
